### PR TITLE
Fix a python3 incompatibility

### PR DESCRIPTION
**Pull request by elabrosseRodeofx**
I spotted a few python 3 incompatibilities.

### CODE REVIEW

File: shotgun_api3/lib/mimetypes.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 import sys
 import posixpath
 import urllib
+import urllib.parse
```
Comment (elabrosseRodeofx) [inline shotgun_api3/lib/mimetypes.py]: urllib.splittype no longer exists in python3. It's been moved to the parse module.

File: shotgun_api3/lib/mimetypes.py [hunk 2]
```diff
@@ -115,7 +117,10 @@ def guess_type(self, url, strict=True):
         Optional `strict' argument when False adds a bunch of commonly found,
         but non-standard types.
         """
-        scheme, url = urllib.splittype(url)
+        parseResult = urlparse(url)
+        scheme = parseResult.scheme
+        url = parseResult.netloc + parseResult.path
```
Comment (elabrosseRodeofx) [inline shotgun_api3/lib/mimetypes.py]: This doesn't give us exactly the same string as the previous implementation.
previous implementation:
```
>>> url = urllib.splittype('file://host/path.ext')[-1]
>>> url 
'//host/path.ext'
```

current implementation:
```
>>> parseResult = six.moves.urllib.parse.urlparse('file://host/path.ext')
>>> url = parseResult.netloc + parseResult.path
>>> url
'host/path.ext'
```

**Comment by JeanChristopheMorinRodeoFX** [comment]
Please make sure you also open a PR in the upstream repo. Let me know if you need help with this.

**Review by JeanChristopheMorinRodeoFX — APPROVED** [review]

**Comment by elabrosseRodeofx** [comment]
@JeanChristopheMorinRodeoFX 
This doesn't need to be merged in the upstream repo since it already includes the fix.
https://github.com/shotgunsoftware/python-api/blob/master/shotgun_api3/lib/mockgun/mockgun.py#L499

**Pull request merged**